metricbeat/module/jolokia/jmx/jmx.go

```diff
@@ -93,24 +92,28 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
-
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	var allEvents []common.MapStr
 
 	allEvents, err := m.jolokia.Fetch(m)
 	if err != nil {
-		return nil, err
+		logger.Error(err)
+		reporter.Error(err)
+		return
 	}
 
 	// Set dynamic namespace.
-	var errs multierror.Errors
 	for _, event := range allEvents {
 		_, err := event.Put(mb.NamespaceKey, m.namespace)
 		if err != nil {
-			errs = append(errs, err)
+			logger.Error(err)
+			reporter.Error(err)
 		}
+		reporter.Event(mb.Event{
+			MetricSetFields: event,
+		})
```

Comment:
I think that you can place the contents of line 110 here in a field called `Namespace` to achieve the same result.